frontend/src/contexts/AssessmentContext.ts

```diff
@@ -1,31 +1,38 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
 
+import type { SetStateAction } from "react";
 import { createContext } from "react";
+import type { LocationDescriptor } from "history";
 
+import type { RedirectableNavigatePromptHistory } from "../components/common/navigation/useRedirectableNavigatePrompt";
 import type { Question } from "../types/QuestionTypes";
 
 type AssessmentContextType = {
+  redirectableHistory: RedirectableNavigatePromptHistory;
+  setQuestionEditorDirty: (_isDirty: SetStateAction<boolean | null>) => void;
```

Comment:
is there a way to set this in `QuestionEditorContext`? and handle the reload prompt logic separately in QuestionEditor

Reply:
I tried for a while and couldn't get it to work....  the problem is that in the component tree, the question editor technically lives underneath the assessment editor, so we have to have some way of disabling the top-level prompt. The alternative would be a refactor to move the question editor into its own component tree, and the effort there doesn't seem worth it to me.

Reply:
There's another problem I forgot about here, which is that `react-dom` prevents you from even creating two `<Prompt>`s at once!